documentation/testing_plans.md

```diff
@@ -266,18 +266,18 @@ is a stub or a mock.
 You can stub or mock the following functions. Click on each function to view its
 documentation.
 
-| plan function | Stub function | Mock function |
-| --- | --- | --- |
-| [`apply`](plan_functions.md#apply) | [`allow_apply`](boltspec_reference.md#allow-apply) | - |
-| [`apply_prep`](plan_functions.md#apply-prep) | [`allow_apply_prep`](boltspec_reference.md#allow-apply-prep) | - |
-| [`download_file`](plan_functions.md#download-file) | [`allow_download`](boltspec_reference.md#allow-download) | [`expect_download`](boltspec_reference.md#expect_download) |
-| [`out::message`](plan_functions.md#outmessage) | [`allow_out_message`](boltspec_reference.md#allow-out-message) | [`expect_out_message`](boltspec_reference.md#expect-out-message) |
-| [`out::verbose`](plan_functions.md#outverbose) | [`allow_out_verbose`](boltspec_reference.md#allow-out-verbose) | [`expect_out_verbose`](boltspec_reference.md#expect-out-verbose) |
-| [`run_command`](plan_functions.md#run-command) | [`allow_command`](boltspec_reference.md#allow-command) | [`expect_command`](boltspec_reference.md#expect-command) |
-| [`run_plan`](plan_functions.md#run-plan) | [`allow_plan`](boltspec_reference.md#allow-plan) | [`expect_plan`](boltspec_reference.md#expect-plan) |
-| [`run_script`](plan_functions.md#run-script) | [`allow_script`](boltspec_reference.md#allow-script) | [`expect_script`](boltspec_reference.md#expect-script) |
-| [`run_task`](plan_functions.md#run-task) | [`allow_task`](boltspec_reference.md#allow-task) | [`expect_task`](boltspec_reference.md#expect-task) |
-| [`upload_file`](plan_functions.md#upload-file) | [`allow_upload`](boltspec_reference.md#allow-upload) | [`expect_upload`](boltspec_reference.md#expect-upload) |
+| plan function | Stub function | Global stub function | Mock function
+| --- | --- | --- | --- |
+| [`apply`](plan_functions.md#apply) | [`allow_apply`](boltspec_reference.md#allow-apply) | - | - |
+| [`apply_prep`](plan_functions.md#apply-prep) | [`allow_apply_prep`](boltspec_reference.md#allow-apply-prep) | - | - |
+| [`download_file`](plan_functions.md#download-file) | [`allow_download`](boltspec_reference.md#allow-download) | allow_any_download | [`expect_download`](boltspec_reference.md#expect_download) |
+| [`out::message`](plan_functions.md#outmessage) | [`allow_out_message`](boltspec_reference.md#allow-out-message) | ['allow_any_message'] | [`expect_out_message`](boltspec_reference.md#expect-out-message) |
+| [`out::verbose`](plan_functions.md#outverbose) | [`allow_out_verbose`](boltspec_reference.md#allow-out-verbose) | ['allow_any_verbose'] | [`expect_out_verbose`](boltspec_reference.md#expect-out-verbose) |
+| [`run_command`](plan_functions.md#run-command) | [`allow_command`](boltspec_reference.md#allow-command) | ['allow_any_command'] | [`expect_command`](boltspec_reference.md#expect-command) |
```

Comment:
@beechtom , I added the other `allow_any_*` that you mentioned in a previous comment. Do you think it would be best to give them their own respective sections in `boltspec_reference` (allow_any_download, allow_any_command, etc...) or would it be better to go to the end of `Global stubs` and put a section for `alloow_any_*` and just list them there?

Reply:
All of the `allow_any_*` functions should be listed under the "Global stubs" section of boltspec_reference.md, similar to `allow_any_plan` and `allow_any_task`.